assets/src/scripts/components/graph/index.spec.js

```diff
@@ -63,14 +63,15 @@ describe('graph component', () => {
                 responseText: MOCK_WATER_LEVEL_RESPONSE,
                 contentType: 'text/xml'
             });
-            spyOn(window, 'requestAnimationFrame').and.callFake(func => func());
-            window.setTimeout(() => {
-                expect(div.classed('loading')).toBe(false);
-                expect(div.classed('has-error')).toBe(false);
-                expect(div.selectAll('.tooltip-text').size()).toEqual(1);
-                expect(div.selectAll('circle.focus').size()).toBe(1);
-                expect(div.selectAll('.line-segment').size()).toBe(4);
-                done();
+            window.requestAnimationFrame(() => {
+                window.requestAnimationFrame(() => {
```

Comment:
These tests were tricky... what kind of error were you getting on Chrome?

Reply:
It looked like the expectations were getting called before the components were rendered. I had to do two because requestAnimationFrame is called prior to rendering.

At any rate the test now passes on Chrome, Firefox, and Headless Chrome

Reply:
This is randomly causing problems for me with the `test:watch` task (using Firefox).... karma watch with Chrome seems okay. Seems this is going to be tricky to completely fix.

Reply:
What version of Firefox are you using? I'm using 63.0 and just ran the tests. It's weird since we both run on Macs.

Reply:
Sorry, I said that backwards... Chrome has the problem, not Firefox. `karma start --debug --no-single-run --auto-watch --browsers Chrome` fails on the first re-run.

Reply:
Still weird. Mine pass on the first re-run (and the second and third). I wonder if it's some sort of configuration that we have different? Have no idea what that might be.